packages/nimble-components/src/table/template.ts

```diff
@@ -79,7 +79,7 @@ export const template = html<Table>`
                                 ${ref('collapseAllButton')}
                                 ${'' /* tabindex managed dynamically by KeyboardNavigationManager */}
                                 tabindex="-1"
-                                class="collapse-all-button ${x => `${x.showCollapseAll ? 'visible' : ''}`}"
+                                class="collapse-all-button ${x => `${x.showCollapseAll ? 'visible' : ''}`} ${x => `${x.canHaveCollapsibleRows ? '' : 'hidden'}`}"
```

Comment:
This, I think, is the only part of the implementation that might merit re-thinking. At a glance it's odd to have one observable property, when true, set the CSS `visible` state, and a separate observable property, when false, set the CSS `hidden` state. I think it would be cleaner if we made a single observable property be enumerated. At the moment the right semantics are eluding me, as the three terms that leapt to mind were `show`, `hide` and `disable`, but the `hide` would map to "", while `disable` would map to `hidden`, which seems strange.